contracts/ownership/Heritable.sol

```diff
@@ -0,0 +1,100 @@
+pragma solidity ^0.4.11;
+
+
+import "./Ownable.sol";
+
+
+/**
+ * @title Heritable
+ * @dev The Heritable contract provides ownership transfer capabilities, in the
+ * case that the current owner stops "heartbeating". Only the heir can pronounce the
+ * owner's death.
+ */
+contract Heritable is Ownable {
+  address public heir;
+
+  // Time window the owner has to notify they are alive.
+  uint public heartbeatTimeout;
+
+  // Timestamp of the owner's death, as pronounced by the heir.
+  uint public timeOfDeath;
+
+
+  event HeirChanged(address indexed owner, address indexed newHeir);
+  event OwnerHeartbeated(address indexed owner);
+  event OwnerProclaimedDead(address indexed owner, address indexed heir, uint timeOfDeath);
+  event HeirOwnershipClaimed(address indexed previousOwner, address indexed newOwner);
+
+
+  /**
+   * @dev Throw an exception if called by any account other than the heir's.
+   */
+  modifier onlyHeir() {
+    require(msg.sender == heir);
+    _;
+  }
+
+
+  /**
+   * @notice Create a new Heritable Contract with heir address 0x0.
+   * @param _heartbeatTimeout time available for the owner to notify they are alive,
+   * before the heir can take ownership.
+   */
+  function Heritable(uint _heartbeatTimeout) public {
+    setHeartbeatTimeout(_heartbeatTimeout);
+  }
+
+  function setHeir(address newHeir) public onlyOwner {
+    require(newHeir != owner);
+    heartbeat();
+    HeirChanged(owner, newHeir);
+    heir = newHeir;
```

Comment:
If we provide a setter, I would go with a `private` visibility for the `heir` and providing a getter as well. This will ensure that inheriting contracts can not modify the `heir` in another way than the one proposed.

Reply:
we may want the same thing for `heartbeatTimeout` and `timeOfDeath`

Reply:
Makes sense, but this is not something we're implementing in general in OZ. Eg. Ownable has:
`address public owner`

Reply:
Actually we are caring about this from now on

Reply:
Great. I won't be changing it in this PR tho.
I will create an issue instead and ping @azavalla